### PR TITLE
Simplify dependency locking and version def

### PR DESCRIPTION
**Pull request by dexamundsen**
- Simplify command for dependency locking by wrapping sub-project tasks within a root task
- Define plugin versions in build.gradle, ensures same version is used across projects
- use Buildscript variables instead of gradle variables. The former is build structure agnostic
- Bump spring boot version from 3.3.3 to 3.3.4; dep management from 1.1.5 to 1.1.6

### CODE REVIEW

File: annotationProcessor/build.gradle [hunk 1]
```diff
@@ -3,8 +3,8 @@ plugins {
 
     // These plugins are needed to resolve versions of spring dependencies
     // transitively added by terra-common-lib
-    id 'org.springframework.boot' version '3.3.3'
-    id 'io.spring.dependency-management' version '1.1.5'
+    id 'io.spring.dependency-management' version "${vSpringDepMgt}"
```
Comment (dexamundsen) [inline annotationProcessor/build.gradle:6]: Versions defined as build script variables in build.gradle.

File: annotationProcessor/build.gradle [hunk 2]
```diff
@@ -39,9 +39,9 @@ dependencies {
         implementation 'com.nimbusds:nimbus-jose-jwt:9.37.3'
     }
 
-    implementation "org.slf4j:slf4j-api:${gradle.vSlf4jApi}"
```
Comment (dexamundsen) [inline annotationProcessor/build.gradle:42]: Build script variables do not need gradle qualifier, hence removed in all references

File: build.gradle [hunk 3]
```diff
@@ -1,22 +1,35 @@
...
+        vSwaggerCodegen = '3.0.57'
+        vPicocliVersion = '4.7.6'
+        vJunit5 = '5.10.2'
+    }
     repositories {
         mavenCentral()
     }
 }
 
-gradle.ext {
```
Comment (dexamundsen) [inline build.gradle:8]: Moved up.

File: build.gradle [hunk 4]
```diff
@@ -1,22 +1,35 @@
 // Required by spotless. https://github.com/diffplug/spotless/issues/747
 buildscript {
+    ext {
+        // plugin versions
+        vSpringBoot = '3.3.4'
```
Comment (dexamundsen) [inline build.gradle:5]: spring boot bumped from 3.3.3 to 3.3.4
dep management plugin bumped from 1.1.5 to 1.1.6

File: build.gradle [hunk 5]
```diff
@@ -1,22 +1,35 @@
...
-    vApacheCommonsText = '1.12.0'
-    vSlf4jApi = '2.0.16'
-    vSpotBugs = '4.8.5'
-    vJackson = '2.17.2'
-    vJersey = '3.1.8' // Java 17 compatible
-    vSwaggerAnnotations = '2.2.22'
-    vSwaggerCodegen = '3.0.57'
-    vPicocliVersion = '4.7.6'
-    vJunit5 = '5.10.2'
+tasks.named('dependencies') {
```
Comment (dexamundsen) [inline build.gradle:29]: This configures a root task dependencies that depends on below subtasks. hence auto triggered.

**Review by tjennison-work — APPROVED** [review]
This is a good improvement. It's surprising to me that Gradle doesn't doesn't have a more automatic way to handle this but I don't have much experience with it.

**Pull request merged**